azure_run/backup.py

```
@@ -1,175 +1,107 @@
+        # Upload to datastore
+        log().info(f"Uploading to datastore {_BACKUP_DATASTORE}:{full_remote_path}...")
+        Dataset.File.upload_directory(local_dir, (dast,full_remote_path))
+        log().info("File upload successful!")
+    except Exception as e:
+        # Clean up and report failure
+        log().error(f"Failed to backup dataset: {name}:{ds.version} to {(dast,full_remote_path)}!")
+        raise e
+    finally:
+        # Clean-up
+        log().debug("Cleaning up temporary data...")
+        with contextlib.suppress(OSError):
```

Ruff (0.8.2)

Undefined name `contextlib`

(F821)